donut/modules/editor/helpers.py

```diff
@@ -0,0 +1,202 @@
+    # some lag ).
+    if res['expired'] >= TIMEOUT:
+        change_lock_status(title, False, forced=True)
+        return False
+    return res['locked']
+
+
+def create_page_in_database(title):
+    """
+    There are some pages that exist but do not have entries in the 
+    database. 
+    """
+    query = """
+    INSERT INTO webpage_files_locks (title) VALUES (%s) ON DUPLICATE KEY UPDATE locked = locked
+    """
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, title)
+        res = cursor.fetchone()
+
+
+def rename_title(oldfilename, newfilename):
+    """
+    Changes the file name of an html file
+    Need to look for paths
+    """
+    oldpath = os.path.join(flask.current_app.root_path,
+                           flask.current_app.config["UPLOAD_WEBPAGES"],
+                           oldfilename + '.md')
+    newpath = os.path.join(flask.current_app.root_path,
+                           flask.current_app.config["UPLOAD_WEBPAGES"],
+                           newfilename + '.md')
+    if os.path.exists(oldpath) and not os.path.exists(newfilename):
+        os.rename(oldpath, newpath)
+
+
+def read_markdown(name):
+    '''
+    Reads in the mark down text from a file.
+    '''
+
+    path = os.path.join(flask.current_app.root_path,
+                        flask.current_app.config['UPLOAD_WEBPAGES'])
+    curFile = read_file(os.path.join(path, name + '.md'))
+    return curFile
+
+
+def read_file(path):
+    '''
+    Reads in a file
+    '''
+    if not os.path.isfile(path):
+        return ''
+
+    with open(path) as f:
+        return f.read()
+
+
+def get_links():
+    '''
+    Get links for all created webpages
+    '''
+    root = os.path.join(flask.current_app.root_path,
+                        flask.current_app.config["UPLOAD_WEBPAGES"])
+    links = glob.glob(root + '/*')
+    results = []
+    for filenames in links:
+        filenames = filenames.replace(root + '/',
+                                      '').replace('.md', '').replace('_', ' ')
+        link = flask.url_for('uploads.display', url=filenames)
+        results.append((link, filenames))
+    return results
+
+
+def remove_link(filename):
+    '''
+    Get rid of matching filenames
+    '''
+    path = os.path.join(flask.current_app.root_path,
+                        flask.current_app.config['UPLOAD_WEBPAGES'])
+    links = glob.glob(path + '/*')
+    for i in links:
+        name = i.replace(path + '/', '').replace('.md', '').replace("_", " ")
+        if filename == name:
+            os.remove(i)
+    query = """DELETE FROM webpage_files_locks WHERE title = %s"""
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, filename)
+
+
+def check_duplicate(filename):
+    """
+    Check to see if there are duplicate file names
+    """
+    path = os.path.join(flask.current_app.root_path,
+                        flask.current_app.config['UPLOAD_WEBPAGES'])
+    links = glob.glob(path + '/*')
```

Comment:
Maybe make a `get_all_uploads()` function to do this globbing?

Reply:
yeah I still think it's more trouble than it's worth since each are a bit different